java/vector/src/main/java/org/apache/arrow/vector/complex/FixedSizeListVector.java

```diff
@@ -496,6 +496,37 @@ public TransferPair makeTransferPair(ValueVector target) {
     return new TransferImpl((FixedSizeListVector) target);
   }
 
+  @Override
+  public int hashCode(int index) {
+    if (isSet(index) == 0) {
+      return 0;
+    }
+    int hash = 0;
+    for (int i = 0; i < listSize; i++) {
+      hash = 31 * vector.hashCode(index * listSize + i);
```

Comment:
do you mean += here?
or maybe hash = 31 * hash + vector.hashCode(index * listSize + i);

Reply:
Thanks, you are right, fixed.